modules/homa-lite/src/lib.rs

```diff
@@ -256,5 +276,304 @@ pub mod module {
 			Self::deposit_event(Event::<T>::XcmDestWeightSet(xcm_dest_weight));
 			Ok(())
 		}
+
+		/// Put in an request to redeem Staking currencies used to mint Liquid currency.
+		/// The redemption will happen after the currencies are unbounded on the relaychain.
+		///
+		/// Parameters:
+		/// -
+		#[pallet::weight(< T as Config >::WeightInfo::request_redeem())]
+		#[transactional]
+		pub fn request_redeem(origin: OriginFor<T>, liquid_amount: Balance, additional_fee: Permill) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			if liquid_amount.is_zero() {
+				// If the maount is zero, cancel previuos redeem request.
+				if let Some((_, _)) = Self::redeem_requests(&who) {
+					// Unreserve the liquid fee and remove the redeem request.
+					ensure!(
+						T::Currency::unreserve(T::LiquidCurrencyId::get(), &who, liquid_amount) == liquid_amount,
+						Error::<T>::InsufficientReservedBalances
+					);
+					RedeemRequests::<T>::remove(&who);
+
+					Self::deposit_event(Event::<T>::RedeemRequestCancelled(who, liquid_amount));
+				}
+			} else {
+				// Put in an redeem request, or add additional amounts to be redeemed
+				let (redeemed_amount, _) = Self::redeem_requests(&who).unwrap_or((0, Permill::zero()));
```

Comment:
this way two transaction is required to reduce redeem amount.
should just replace the current request